### PR TITLE
Be able to cancel image.decode operations

### PR DESCRIPTION
**Pull request by fredj**
Thanks @ahocevar  for the `hasListener` idea !

### CODE REVIEW

File: src/ol/Image.js [hunk 1]
```diff
@@ -155,8 +157,27 @@ class ImageWrapper extends ImageBase {
 export function listenImage(image, loadHandler, errorHandler) {
   const img = /** @type {HTMLImageElement} */ (image);
   if (img.decode) {
-    img.decode().then(loadHandler).catch(errorHandler);
-    return null;
+    // create fake event target to be able to cancel the image decoding.
+    const target = new EventTarget();
```
Comment (ahocevar) [inline src/ol/Image.js]: I think this is a bit overkill. Why not do something like
```diff
diff --git a/src/ol/Image.js b/src/ol/Image.js
index 45d0cf07c..283427747 100644
--- a/src/ol/Image.js
+++ b/src/ol/Image.js
@@ -155,7 +155,15 @@ class ImageWrapper extends ImageBase {
 export function listenImage(image, loadHandler, errorHandler) {
   const img = /** @type {HTMLImageElement} */ (image);
   if (img.decode) {
-    img.decode().then(loadHandler).catch(errorHandler);
+    img.decode().then(function() {
+      if (!img.getAttribute('data-ol-ignore')) {
+        loadHandler();
+      }
+    }).catch(function() {
+      if (!img.getAttribute('data-ol-ignore')) {
+        errorHandler();
+      }
+    });
     return null;
   } else {
     const listenerKeys = [
@@ -174,6 +182,8 @@ export function unlistenImage(listenerKeys) {
   if (listenerKeys) {
     listenerKeys.forEach(unlistenByKey);
     listenerKeys = null;
+  } else {
+    this.getImage().setAttribute('data-ol-ignore', true);
   }
 }
```

Comment (ahocevar) [reply src/ol/Image.js]: Or even nicer: make listenImage() a member function, then you can use the instance to store a flag when the image is not being needed any more.

**Comment by tschaub** [comment]
I didn't see this before putting up #9558.  I think we can simplify things by just returning an `unlisten` function in all cases.

**Comment by fredj** [comment]
#9558

**Pull request closed without merging**